ledger/simulation/simulator.go

```diff
@@ -189,6 +190,18 @@ func (s Simulator) simulateWithTracer(txgroup []transactions.SignedTxn, tracer l
 		return nil, err
 	}
 
+	// check that the extra budget is not exceeding simulation extra budget limit
+	if overrides.ExtraAppBudget != nil && *overrides.ExtraAppBudget > MaxExtraBudget {
+		return nil,
+			EvalFailureError{
```

Comment:
As you're probably aware, if you use `EvalFailureError`, we will catch this error, thread it into our response, and still return HTTP 200. I think HTTP 400 is more appropriate for this, so I would suggest creating an `InvalidTxGroupError`, for which the handler has code to catch and return 400.

Though maybe we should rename `InvalidTxGroupError` to something like `InvalidRequestError`?

(The error situation could use more documentation to make the behavior of these types more clear. Also, I don't think we need `SimulatorError` at the moment.)

Reply:
I support this, it isn't the evaluation that failed, its that too much was requested.

Reply:
ok, I will rename `InvalidTxGroupError` to `InvalidRequestError`, for it encapsulates tx-group error + the over extra budget error we have here.

Reply:
see [omitempty, InvalidRequestError](https://github.com/algorand/go-algorand/pull/5354/commits/cece25744c9a5a7f168ec8ec35711b2b1c94706b)